### PR TITLE
0.6 pytest

### PR DESCRIPTION
**Pull request by SebKuzminsky**
This PR removes the old bit-rotted manual tests and starts adding new automatic tests using pytest.  Run the tests with "make pytest" (which is also included in "make test").

Thanks to @zancas for help with python and the pytest framework.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,7 +16,7 @@ PYTHON_EXE ?= python
 # (introduced in python 2.6)
 DISTUTILS_PLAT_NAME = $(shell $(PYTHON_EXE) setup.py --help build_ext \
 		      | grep -q -- "--plat-name" && echo "--plat-name win32")
-PYTHON_CHECK_STYLE_TARGETS = pycam Tests pyinstaller/hooks/hook-pycam.py scripts/pycam setup.py
+PYTHON_CHECK_STYLE_TARGETS = pycam pyinstaller/hooks/hook-pycam.py scripts/pycam setup.py
```
Comment (zancas) [inline Makefile]: OK, I see what we're doing here, and maybe this isn't the right venue, BUT...

A few thoughts:

   *  do we still need to support svn?
   *  can we start migrating test/build/deployment infrastructure from make, into the `.travis.yml`?
   *  how are we populating this list of targets?  Is it growing as we start managing more elements of the codebase with style checks?

File: Makefile [hunk 2]
```diff
@@ -81,7 +81,12 @@ upload:
 	svn import "$(ARCHIVE_DIR)/$(EXPORT_WIN32)" "$(REPO_TAGS)/archives/$(EXPORT_WIN32)" \
 		-m "added released win32 installer for version $(VERSION)"
 
-test: check-style
+test: check-style pytest
+
+pytest:
+	for TEST in $$(find tests/ -name '*.py'); do \
+	    python -m pytest -v $$TEST; \
+	done
```
Comment (zancas) [inline Makefile]: pytest has builtin logic for traversing down from the directory it receives as an argument.  For each directory in the target recurse into that directory and collect all of the "tests" within files named:  `test_*.py`   (also perhaps other files that include the word "test" in their name).
So, we don't need to manually feed `pytest` each test target.   We should be able to simply say:

`pytest -v ${PROJECTROOT}`

File: tests/Geometry/Polygon.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+from pycam.Geometry.Polygon import *
+from pycam.Geometry.Line import *
```
Comment (zancas) [inline tests/Geometry/Polygon.py]: Alright, dagnabbit, I am going to get to the root of this relative import shenaniganry.   It should "Just Work".

**Comment by zancas** [comment]
@SebKuzminsky Are these pytest results consistent with yours?

pytestresults

File: Makefile (hunk 2 above)
Comment (zancas) [reply Makefile]: Oh...   huh...   pytest didn't pick up the "tests" tests...
...

Comment (zancas) [reply Makefile]: Ahh...   let's `mv tests/Geometry/Polygon.py tests/Geometry/test_Polygon.py`

Or maybe `Test_Polygon.py` testing...

Comment (zancas) [reply Makefile]: @SebKuzminsky your choice, I think it's sufficient to start the module name with "test_" and I _think_ it's case insensitive.

**Comment by zancas** [comment]
I vote we do not alter the Makefile at all.
Instead, I think we should alter `.travis.yml` to run:
`py.test .`

I am not yet sure what to do about the `test_intersection.py` failures, are they new?

**Comment by SebKuzminsky** [comment]
Oops, i missed the pycam/Tests directory.  I'll move the Polygon tests there.  And I'll change the name to begin with "test_" for consistency with the existing tests, and to make pytest happy.  And that lets me simplify the Makefile test target...

**Comment by SebKuzminsky** [comment]
The intersection tests have never passed.  They failed when they were first committed to the "test" branch on SourceForge back in 2013 and they failed when the "test" branch was merged into master in February 2017.  I'm going to disable them.

**Comment by SebKuzminsky** [comment]
Grumble, on Debian 9 (Stretch, aka stable), python-pytest supplies /usr/bin/pytest, but on Ubuntu 14.04 (Trusty, where our Travis instance runs) it does not.  On Trusty it provides /usr/bin/py.test, but it doesn't understand the skip directives for the failing tests.

**Comment by SebKuzminsky** [comment]
@zancas I chose to update the "make test" target to run pytest (in addition to the flake check it was running before) because I like being abe to run all of a project's tests with a single unchanging command.  I think "make test" is a fairly common way to do this.  It's nice because it's independent of implementation details like whether it's using pytest or something else (or multiple test frameworks).

What's the argument against having a "make test" target?

**Comment by SebKuzminsky** [comment]
All the issues that @zancas identified have been addressed, but all the iterating has left the history of this branch messy.  I've cleaned up the history in a new branch called 0.6-pytest-2, and opened PR #85 to merge it into stable/0.6.  I'm closing this PR.

**Pull request closed without merging**